node/network/collator-protocol/src/lib.rs

```diff
@@ -87,28 +91,26 @@ impl CollatorProtocolSubsystem {
 		Context: SubsystemContext<Message = CollatorProtocolMessage>,
 	{
 		match self.protocol_side {
-		    ProtocolSide::Validator => validator_side::run(ctx, REQUEST_TIMEOUT).await,
-		    ProtocolSide::Collator(id) => collator_side::run(ctx, id).await,
+		    ProtocolSide::Validator(metrics) => validator_side::run(
+				ctx,
+				REQUEST_TIMEOUT,
+				metrics,
+			).await,
+		    ProtocolSide::Collator(id, metrics) => collator_side::run(
+				ctx,
+				id,
+				metrics,
+			).await,
 		}
 	}
 }
 
-/// Collator protocol metrics.
-#[derive(Default, Clone)]
-pub struct Metrics;
-
-impl metrics::Metrics for Metrics {
-	fn try_register(_registry: &prometheus::Registry) 
-		-> std::result::Result<Self, prometheus::PrometheusError> {
-		Ok(Metrics)
-	}
-}
-
 impl<Context> Subsystem<Context> for CollatorProtocolSubsystem
 where
 	Context: SubsystemContext<Message = CollatorProtocolMessage> + Sync + Send,
 {
-	type Metrics = Metrics;
+	// The actual `Metrics` type depends on whether we're on the collator or validator side.
+	type Metrics = ();
```

Comment:
I don't really understand why we expose the `Metrics` type in the trait if it's ok to just fake it and use a different type internally. I assume this is fine as @ordian has already approved this PR, but it may be worth considering whether we could remove `type Metrics` from `trait Subsystem`.

If on the other hand there's a reason to publish this type, then we really should be distributing it via an enum.